homeassistant/components/xiaomi_aqara/config_flow.py

```diff
@@ -0,0 +1,167 @@
+            xiaomi = XiaomiGatewayDiscovery(self.hass.add_job, [], self.interface)
+            await self.hass.async_add_executor_job(xiaomi.discover_gateways)
+            self.gateways = xiaomi.gateways
+
+            # if host is already known by zeroconf discovery
+            if self.host is not None:
+                self.selected_gateway = self.gateways.get(self.host)
+                if self.selected_gateway is not None:
+                    return await self.async_step_settings()
+
+                errors["base"] = "not_found_error"
+            else:
+                if len(self.gateways) == 1:
+                    self.selected_gateway = list(self.gateways.values())[0]
+                    return await self.async_step_settings()
+                if len(self.gateways) > 1:
+                    return await self.async_step_select()
+
+                errors["base"] = "discovery_error"
+
+        return self.async_show_form(
+            step_id="user", data_schema=GATEWAY_CONFIG, errors=errors
+        )
+
+    async def async_step_select(self, user_input=None):
+        """Handle multiple aqara gateways found."""
+        errors = {}
+        if user_input is not None:
+            ip_adress = user_input["select_ip"]
+            self.selected_gateway = self.gateways[ip_adress]
+            return await self.async_step_settings()
+
+        select_schema = vol.Schema(
+            {
+                vol.Required("select_ip"): vol.In(
+                    [gateway.ip_adress for gateway in self.gateways.values()]
+                )
+            }
+        )
+
+        return self.async_show_form(
+            step_id="select", data_schema=select_schema, errors=errors
+        )
+
+    async def async_step_zeroconf(self, discovery_info):
+        """Handle zeroconf discovery."""
+        name = discovery_info.get("name")
+        self.host = discovery_info.get("host")
+        mac_address = discovery_info.get("properties", {}).get("mac")
+
+        if not name or not self.host or not mac_address:
+            return self.async_abort(reason="not_xiaomi_aqara")
+
+        # Check if the discovered device is an xiaomi aqara gateway.
+        if not name.startswith(ZEROCONF_GATEWAY):
+            _LOGGER.debug(
+                "Xiaomi device '%s' discovered with host %s, not identified as xiaomi aqara gateway",
+                name,
+                self.host,
+            )
+            return self.async_abort(reason="not_xiaomi_aqara")
+
+        # format mac (include semicolns and make uppercase)
+        mac_address = format_mac(mac_address)
+
+        unique_id = mac_address
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured({CONF_HOST: self.host})
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update({"title_placeholders": {"name": self.host}})
+
+        return await self.async_step_user()
```

Comment:
Do we still need to ask the user for interface if we know the host address of the gateway?

Reply:
yes because the interface is the address the socket binds to in order to receive messages from the gateway. that is not the gateway ip, but the ip of the device that is running HomeAssistant.
Or 'any', however 'any' does not always work (for me when I have VPN enabled on my dev PC 'any" does not work, withouth VPN it does work.